tests/test_metadata/test_v3.py

```diff
@@ -304,6 +304,19 @@ def test_metadata_to_dict(
     assert observed == expected
 
 
+def test_json_indent():
+    m = GroupMetadata()
+    d = m.to_buffer_dict(default_buffer_prototype())["zarr.json"].to_bytes()
+    class TestIndentEncoder(json.JSONEncoder):
+        def __init__(self, *args: Any, **kwargs: Any) -> None:
+            super().__init__(*args, **kwargs)
+            self.indent = 2
+
+    # expected has extra ' ' on each line compared with json.dumps( indent=2)
+    expected = json.dumps(json.loads(d), cls=TestIndentEncoder).encode()
```

Comment:
Would it be possible to refactor this test so that it tests that the `json_indent` can be changed at runtime to produce different JSON-formatted strings? e.g., something like 
```python
@pytest.mark.parametrize('indent', (2,4))
def test_json_indent(indent: int):
    with config.set({"json_indent": indent}):
     #    ... write the metadata document to string 
    # compare with expected  
```

Reply:
a separate test could check that the default `json_indent` value is actually 2

Reply:
I think you have that in test_config.py https://github.com/zarr-developers/zarr-python/blob/ed0f199e8542b410ae655c8168a49a0be6237cd4/tests/test_config.py#L47 which you added in #1952 ?

Reply:
quite right! no need for anything new here